runners/google-cloud-dataflow-java/worker/src/test/java/org/apache/beam/runners/dataflow/worker/streaming/WeightBoundedQueueTest.java

```diff
@@ -0,0 +1,231 @@
+                throw new RuntimeException(e);
+              }
+            });
+
+    pollThread.start();
+    Thread.sleep(pollWaitTimeMillis / 100);
+    queue.put(insertedValue1);
+    pollThread.join();
+
+    assertEquals(insertedValue1, pollResult.get());
+  }
+
+  @Test
+  public void testPoll_withLargeTimeout() throws InterruptedException {
+    WeightedBoundedQueue<Integer> queue =
+        WeightedBoundedQueue.create(MAX_WEIGHT, i -> Math.min(MAX_WEIGHT, i));
+    int insertedValue1 = 1;
+
+    AtomicInteger pollResult = new AtomicInteger();
+    Thread pollThread =
+        new Thread(
+            () -> {
+              int polled;
+              try {
+                polled = queue.poll(1, TimeUnit.MINUTES);
+                pollResult.set(polled);
+              } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+              }
+            });
+
+    pollThread.start();
+    Thread.sleep(30000); // 30 sec.
+    queue.put(insertedValue1);
+    pollThread.join();
+
+    assertEquals(insertedValue1, pollResult.get());
+  }
+
+  @Test
+  public void testPoll_withTimeout_timesOut() throws InterruptedException {
+    WeightedBoundedQueue<Integer> queue =
+        WeightedBoundedQueue.create(MAX_WEIGHT, i -> Math.min(MAX_WEIGHT, i));
+    int defaultPollResult = -10;
+    int pollWaitTimeMillis = 100;
+    int insertedValue1 = 1;
+
+    // AtomicInteger default isn't null, so set it to a negative value and verify that it doesn't
+    // change.
+    AtomicInteger pollResult = new AtomicInteger(defaultPollResult);
+
+    Thread pollThread =
+        new Thread(
+            () -> {
+              int polled;
+              try {
+                polled = queue.poll(pollWaitTimeMillis, TimeUnit.MILLISECONDS);
+                pollResult.set(polled);
+              } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+              }
+            });
+
+    pollThread.start();
+    Thread.sleep(pollWaitTimeMillis * 100);
+    queue.put(insertedValue1);
+    pollThread.join();
+
+    assertEquals(defaultPollResult, pollResult.get());
+  }
+
+  @Test
+  public void testPoll_emptyQueue() {
+    WeightedBoundedQueue<Integer> queue =
+        WeightedBoundedQueue.create(MAX_WEIGHT, i -> Math.min(MAX_WEIGHT, i));
+
+    assertNull(queue.poll());
+  }
+
+  @Test
+  public void testTake() throws InterruptedException {
+    WeightedBoundedQueue<Integer> queue =
+        WeightedBoundedQueue.create(MAX_WEIGHT, i -> Math.min(MAX_WEIGHT, i));
+
+    AtomicInteger value = new AtomicInteger();
+    // Should block until value is available
+    Thread takeThread =
+        new Thread(
+            () -> {
+              try {
+                value.set(queue.take());
+              } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+              }
+            });
+    takeThread.start();
+
+    Thread putThread =
```

Comment:
remove putThread, can just be inlined to test.

Reply:
done